rfcs/0001-tide-roadmap.md

```diff
@@ -0,0 +1,88 @@
+# Detailed description
+[detailed-description]: #detailed-description
+
+# Tide
+[tide]: #tide
+
+The project has seen a lot of discussion happen in [issues][issues] as well as the discord channel. The general conversation currently revolves around getting the core framework stabilized in order to build features on top of that. To this end, the currently [open PR by @aturon][context-pr] works to refactor the way data is passed along to endpoint functions. This change is a major change allowing data extraction to be more explicit.
+
+ Once this feature lands, here are some of the issues that work can begin on:
+
+ * [Authentication in Tide][issues-99]
+ * [Serving static files][issues-63]
+ * [URL generation][issues-24]
+
+There has been some discussion on the [Sprint 1 goals issue][sprint-goals], and a really good overview for the features to work on has been been [outlined by @gruberb][goals-outline]. This outline a great basis to set up the roadmap for Tide. It consists of the following broad goals:
+
+* [ ] Stabilize tide core.
+* [ ] Session management
+* [ ] Authentication
+
+## Stabilize tide core
+[stabilize-tide-core]: #stabilize-tide-core
+
+Currently [the PR][context-pr] is open, with most of the core changes done. There is ongoing discussion regarding the change, and you can follow with the progress directly on the PR.
+
+
+### Goals
+[stabilize-core-goals]: #stabilize-core-goals
+
+* [ ] Merge PR
+* [ ] Resolve design questions.
+
+## Session management
+[session-management]: #session-management
+
+Current discussion around session management in tide is centered around the [design issue][issues-9] for the same. @tomhoule has got a working generic session implementation written against the new tide core changes. You can check the [project out here][session-project]. This provides types to define middleware and custom session storage backends which hook directly into the `Context` object provided to the endpoint function.
+
+The current discussion is centered on providing a simple in memory session storage with Cookies as the default in the framework, and provide external crates to hook into external data stores.
+
+This change is currently blocked by the tide core changes.
+
+### Goals
+
+* [ ] Stable API.
+* [ ] In memory session backend
+
+## Authentication
+[Authentication]: #authentication
+
+Currently rust web frameworks have disparate ways of authentication. One of the goals of the Async ecosystem WG is to provide a common set of crates which can be used to build higher level abstractions. In terms of Tide, there currently isn't an authentication story present. @tomhoule has built cookie based authentication into the session management middleware as a POC. Therefore the goals for this sprint would be to sketch out the design of the authentication API for tide.
+
+### Goals
+
+* [ ] Design authentication API in Tide.
+
+# Areweasyncyet
+[Areweasyncyet]: #areweasyncyet
+
+TODO
+
+# Arewewebyet
+[Arewewebyet]: #arewewebyet
+
+TODO
```

Comment:
Just a sentence about pruning old content, and adding new content is probably enough. Something like:

> "Prune outdated content from arewewebyet, and update resources."